backend/beyond_the_loop/services/credit_service.py

```
@@ -0,0 +1,287 @@
+            
+        Returns:
+            dict: Information about the payment intent
+            
+        Raises:
+            HTTPException: If there's an error with the payment or recharge process
+        """
+        try:
+            company = Companies.get_company_by_id(user.company_id)
+            
+            if not company.stripe_customer_id:
+                raise HTTPException(status_code=400, detail="No payment method found. Please add a payment method first.")
```

Ruff (0.13.1)

Abstract `raise` to an inner function

(TRY301)